Source/Dafny/Compilers/Compiler-python.cs

```diff
@@ -898,19 +898,12 @@ protected class ClassWriter : IClassWriter {
     protected override void EmitNewArray(Type elmtType, IToken tok, List<Expression> dimensions, bool mustInitialize,
         ConcreteSyntaxTree wr, ConcreteSyntaxTree wStmts) {
       var initValue = mustInitialize ? DefaultValue(elmtType, wr, tok, true) : "None";
-      if (dimensions.Count == 1) {
-        // handle the common case of 1-dimensional arrays separately
-        wr.Write($"{DafnyArrayClass}([{initValue} for _ in range");
-        TrParenExpr(dimensions[0], wr, false, wStmts);
-        wr.Write("])");
-      } else {
-        wr.Write($"{DafnyRuntimeModule}.newArray({initValue}");
-        foreach (var dim in dimensions) {
-          wr.Write(", ");
-          TrExpr(dim, wr, false, wStmts);
-        }
-        wr.Write(")");
+      wr.Write($"{DafnyArrayClass}({initValue}");
+      foreach (var dim in dimensions) {
+        wr.Write(", ");
+        TrExpr(dim, wr, false, wStmts);
```

Comment:
I imagine this would have come up somewhere else, but how do we guarantee that this TrExpr has enough parentheses to not capture the commas around it?

Reply:
I can't come up with something sufficiently ambiguous that we would actually generate.